sale_promotion_discount_in_field/models/sale_order.py

```diff
@@ -0,0 +1,131 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _create_new_no_code_promo_reward_lines(self):
+        """
+        Override core method to save program with new discount apply rule
+        """
+        self.ensure_one()
+        order = self
+        programs = order._get_applicable_no_code_promo_program()
+        programs = (
+            programs._keep_only_most_interesting_auto_applied_global_discount_program()
+        )
+        if any([program.reward_type == "discount_line" for program in programs]):
+            for program in programs:
+                error_status = program._check_promo_code(order, False)
+                if not error_status.get("error"):
+                    if program.promo_applicability == "on_next_order":
+                        order.state != "cancel" and order._create_reward_coupon(program)
+                    elif (
+                        program.discount_line_product_id.id
+                        not in self.order_line.mapped("product_id").ids
+                    ):
+                        if program.reward_type == "discount_line":
+                            self._get_reward_line_values(program)
+                        else:
+                            self.write(
+                                {
+                                    "order_line": [
+                                        (0, False, value)
+                                        for value in self._get_reward_line_values(
+                                            program
+                                        )
+                                    ]
+                                },
+                            )
+                    order.no_code_promo_program_ids |= program
+        else:
+            super()._create_new_no_code_promo_reward_lines()
+
+    def _get_reward_line_values(self, program):
+        """
+        Overrider core method to set discount for line if program has
+        'discount_line' reward type
+        """
+        if program.reward_type == "discount_line":
+            self._set_reward_discount_for_lines(program)
+            return []
+        else:
+            return super()._get_reward_line_values(program)
```

Comment:
for instance
```suggestion
        if program.reward_type != "discount_line":
            return super()._get_reward_line_values(program)
        self._set_reward_discount_for_lines(program)
        return []           
```
Hmm here you are writing in a method that is supposed to "get" something?